### ISSUE TITLE
[BUG] library install error: Fail to build the dependencie "maturin" 

### ISSUE THREAD

**Issue by FloWIsCoding** [interactions-py/interactions.py#388]
I'm not able to install discord-py-interactions because of one of its dependencies named "maturin" 

**Reproducing steps**
1. `pip install -U discord-py-interactions`
4. See the  error given in the terminal or logger file.

**What actually happened**
I received this error given from my terminal:

```
Installing build dependencies ... error
  ERROR: Command errored out with exit status 1:
   command: /data/data/com.termux/files/usr/bin/python3 /data/data/com.termux/files/usr/tmp/pip-standalone-pip-h_ffi456/__env_pip__.zip/pip install --ignore-installed --no-user --prefix /data/data/com.termux/files/usr/tmp/pip-build-env-_meffbx2/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'maturin>=0.11.2,<0.12'
       cwd: None
  Complete output (57 lines):
  Collecting maturin<0.12,>=0.11.2
    Downloading maturin-0.11.5.tar.gz (469 kB)
    Installing build dependencies: started
    Installing build dependencies: finished with status 'done'
    Getting requirements to build wheel: started
    Getting requirements to build wheel: finished with status 'done'
    Preparing metadata (pyproject.toml): started
    Preparing metadata (pyproject.toml): finished with status 'done'
  Collecting toml~=0.10.0
    Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
  Building wheels for collected packages: maturin
    Building wheel for maturin (pyproject.toml): started
    Building wheel for maturin (pyproject.toml): finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /data/data/com.termux/files/usr/bin/python3 /data/data/com.termux/files/usr/tmp/tmpuw6zo7yw_in_process.py build_wheel /data/data/com.termux/files/usr/tmp/tmpi4kfraq5
         cwd: /data/data/com.termux/files/usr/tmp/pip-install-1lkl27vl/maturin_419dcf7369ae4d5fb8aa6b32e05dff50
    Complete output (36 lines):
    running bdist_wheel
    running build
    installing to build/bdist.linux-aarch64/wheel
    running install
    Traceback (most recent call last):
      File "/data/data/com.termux/files/usr/tmp/tmpuw6zo7yw_in_process.py", line 363, in <module>
        main()
      File "/data/data/com.termux/files/usr/tmp/tmpuw6zo7yw_in_process.py", line 345, in main
        json_out['return_val'] = hook(**hook_input['kwargs'])
      File "/data/data/com.termux/files/usr/tmp/tmpuw6zo7yw_in_process.py", line 261, in build_wheel
        return _build_backend().build_wheel(wheel_directory, config_settings,
      File "/data/data/com.termux/files/usr/tmp/pip-build-env-tw7kfyv2/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 216, in build_wheel
        return self._build_with_temp_dir(['bdist_wheel'], '.whl',
      File "/data/data/com.termux/files/usr/tmp/pip-build-env-tw7kfyv2/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 202, in _build_with_temp_dir
        self.run_setup()
      File "/data/data/com.termux/files/usr/tmp/pip-build-env-tw7kfyv2/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 145, in run_setup
        exec(compile(code, __file__, 'exec'), locals())
      File "setup.py", line 106, in <module>
        setup(
      File "/data/data/com.termux/files/usr/tmp/pip-build-env-tw7kfyv2/overlay/lib/python3.10/site-packages/setuptools/__init__.py", line 153, in setup
        return distutils.core.setup(**attrs)
      File "/data/data/com.termux/files/usr/lib/python3.10/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/data/data/com.termux/files/usr/lib/python3.10/distutils/dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "/data/data/com.termux/files/usr/lib/python3.10/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/data/data/com.termux/files/usr/tmp/pip-build-env-tw7kfyv2/overlay/lib/python3.10/site-packages/wheel/bdist_wheel.py", line 335, in run
        self.run_command('install')
      File "/data/data/com.termux/files/usr/lib/python3.10/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/data/data/com.termux/files/usr/lib/python3.10/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "setup.py", line 58, in run
        raise RuntimeError(
    RuntimeError: cargo not found in PATH. Please install rust (https://www.rust-lang.org/tools/install) and try again
    ----------------------------------------
    ERROR: Failed building wheel for maturin
  Failed to build maturin
  ERROR: Could not build wheels for maturin, which is required to install pyproject.toml-based projects
```

**Versions and device**
- i tried on Python 3.8, Python 3.9 and finally on Python 3.10. 
- i use a phone on Android

**Comment by V3ntus** [comment]
Have you tried installing rust like what the error states?

And please title your issue accordingly :)

**Comment by Catalyst4222** [comment]
My best guess is that it's something to do with you using an android, since some other errors I've found like this happen on ARM processors (just a guess though)
First try `pip install --upgrade pip setuptools wheel`, and if it doesn't work after that, you'll have to do something with rust

**Comment by FloWIsCoding** [comment]
I tried but the error still here

**Comment by FloWIsCoding** [comment]
> Have you tried installing rust like what the error states?
> 
> And please title your issue accordingly :)

Excuse me, is rust a python library ?

**Comment by V3ntus** [comment]
> I tried but the error still here

Can you make sure pip can find `cargo`? Try `which cargo`. If empty, try `pkg install rust cargo`

> Excuse me, is rust a python library ?

As Catalyst states, no, it is an external, standalone programming language and compiler, which must be installed separately, not from pip as pip is the package manager for Python

**Comment by Catalyst4222** [comment]
> Excuse me, is rust a python library?

Rust is another programming language, which is what make this error weird

**Comment by FloWIsCoding** [comment]
> > I tried but the error still here
> 
> Can you make sure pip can find `cargo`? Try `which cargo`. If empty, try `pkg install rust cargo`
> 
> > Excuse me, is rust a python library ?
> 
> As Catalyst states, no, it is an external, standalone programming language and compiler, which must be installed separately, not from pip as pip is the package manager for Python

So I just installed Rust and a new error appeared

**Comment by V3ntus** [comment]
@FloWIsCoding and what's the error?

**Comment by FloWIsCoding** [comment]
@V3ntus I'm actually using my computer so I don't have the error anymore. Thank you

**Comment by FayeDel** [comment]
Hiya!
Closing this issue by the way.

Just for hindsight reasons, this error popped up because of a requirement for your Android app, `termux`, which needs maturin, which haphazardly requires rust and complicates installing this library (I don't know a further reason why, I don't have an Android app to test this);
tl;dr it was an environment issue and not specifically this library on installation
